tests/solvers/quadratic/ht_test.py

```diff
@@ -0,0 +1,129 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+import pytest
+from ott.geometry import geometry, pointcloud
+from ott.geometry.costs import PNormP
+from ott.problems.quadratic import quadratic_problem
+from ott.solvers.linear import implicit_differentiation as implicit_lib
+from ott.solvers.linear import sinkhorn
+from ott.solvers.quadratic import histogram_transport
+
+
+class TestHistogramTransport:
+
+  @pytest.fixture(autouse=True)
+  def initialize(self, rng: jax.random.PRNGKeyArray):
+    d_x = 2
+    d_y = 3
+    self.n, self.m = 6, 7
+    rngs = jax.random.split(rng, 6)
+    self.x = jax.random.uniform(rngs[0], (self.n, d_x))
+    self.y = jax.random.uniform(rngs[1], (self.m, d_y))
+    a = jax.random.uniform(rngs[2], (self.n,)) + 1e-1
+    b = jax.random.uniform(rngs[3], (self.m,)) + 1e-1
+    self.a = a / jnp.sum(a)
+    self.b = b / jnp.sum(b)
+    self.cx = jax.random.uniform(rngs[4], (self.n, self.n))
+    self.cy = jax.random.uniform(rngs[5], (self.m, self.m))
+    self.tau_a = 0.8
+    self.tau_b = 0.9
+
+  @pytest.mark.fast()
+  def test_ht_pointcloud(self):
+    """Test basic computations point clouds."""
+    geom_x = pointcloud.PointCloud(self.x)
+    geom_y = pointcloud.PointCloud(self.y)
+    tau_a, tau_b = (1.0, 1.0)
+    prob = quadratic_problem.QuadraticProblem(
+        geom_x, geom_y, a=self.a, b=self.b, tau_a=tau_a, tau_b=tau_b
+    )
+    solver = histogram_transport.HistogramTransport(
+        epsilon=1e-1,
+        min_iterations=100,
+        max_iterations=100,
+        epsilon_sort=-1,
+        cost_fn=PNormP(2.0),
+        method="quantile",
```

Comment:
I think this function should be parametrized by:
- different `epsilon_sort`s
- different methods
- different cost functions